### PR TITLE
chore: add e2e test for double encoding of query parameters

### PR DESCRIPTION
**Pull request by kulgg**
_What issues does this PR close?_

```[tasklist]
### Checklist
- [x] The PR title is short and expressive.
- [x] I have updated the CHANGELOG.md. See [changelog_update.md](https://github.com/sovity/edc-extensions/tree/main/docs/dev/changelog_updates.md) for more information.
- [x] I have updated the Deployment Migration Notes Section in the CHANGELOG.md for any configuration / external API changes.
- [x] I have performed a **self-review**
```

### CODE REVIEW

File: tests/src/test/java/de/sovity/edc/e2e/DataSourceQueryParamsTest.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                .contractAgreementId(contractAgreementId)
+                .dataSinkProperties(dataAddress.getDataSinkProperties())
+                .build();
+        consumerClient.uiApi().initiateTransfer(transferRequest);
+    }
+
+    private String getProtocolEndpoint(ConnectorRemote connector) {
+        return connector.getConfig().getProtocolEndpoint().getUri().toString();
+    }
+}
```
Comment (richardtreier) [inline tests/src/test/java/de/sovity/edc/e2e/DataSourceQueryParamsTest.java:209]: I am having some trouble validating this, could we have the test cases like this?

```java
var inputQueryString = "a=b"; // "a=b" encoded
var expectedDataSourceReceivedQueryString = "a=b"; // "a=b" double encoded (?)

createAsset(inputQueryString);
createPolicy();
createContractDefinitinon();
negotiate...
transfer...

assertThat(actualReceivedDataSourceQueryString).isEqualTo(expectedDataSourceReceivedQueryString )
```

File: extensions/test-backend-controller/src/main/java/de/sovity/edc/extension/testbackendcontroller/TestBackendController.java [hunk 2]
```diff
@@ -45,4 +47,11 @@ public void setDataSinkValue(String incomingData) {
     public String echoForDataSource(@QueryParam("data") String message) {
         return message;
     }
+
+    @GET
+    @Path("/data-source-params")
```
Comment (richardtreier) [inline extensions/test-backend-controller/src/main/java/de/sovity/edc/extension/testbackendcontroller/TestBackendController.java:52]: data source was the name of the endpoint, we could do it like that `/data-source/echo-query-params`

**Review by richardtreier — CHANGES_REQUESTED** [review]

File: tests/src/test/java/de/sovity/edc/e2e/DataSourceQueryParamsTest.java (hunk 1 above)
Comment (kulgg) [reply tests/src/test/java/de/sovity/edc/e2e/DataSourceQueryParamsTest.java:209]: I would suggest adding a data source spy URL and to keep using the preexisting `validateDataTransferred` function that uses RestAssured to validate the return of that endpoint

**Review by richardtreier — APPROVED** [review]
LGTM

**Pull request merged**